apps/challenges/views.py

```diff
@@ -340,3 +358,221 @@ def challenge_phase_split_list(request, challenge_pk):
+                                                                image))
+        if os.path.isfile(challenge_image_path):
+            challenge_image = open(challenge_image_path, 'rb')
+            challenge_image_file = File(challenge_image)
+        else:
+            challenge_image_file = None
+    else:
+        challenge_image_file = None
+
+    try:
+        with transaction.atomic():
+            serializer = ZipChallengeSerializer(data=yaml_file_data,
+                                                context={'request': request,
+                                                         'challenge_host_team': challenge_host_team})
+            if serializer.is_valid():
+                challenge = serializer.save()
+                challenge_obj = Challenge.objects.get(pk=challenge.pk)
+                if challenge_image_file is not None:
+                    challenge_obj.image.save(image, challenge_image_file, save=True)
+                challenge_obj.evaluation_script.save(evaluation_script,
+                                                     challenge_evaluation_script_file,
+                                                     save=True)
+
+            # Create Leaderboard
+            yaml_file_data_of_leaderboard = yaml_file_data['leaderboard']
+            leaderboard_ids = []
+            for data in yaml_file_data_of_leaderboard:
+                serializer = LeaderboardSerializer(data=data)
+                if serializer.is_valid():
+                    leaderboard = serializer.save()
+                    leaderboard_ids.append(leaderboard.pk)
+
+            # Create Challenge Phase
+            yaml_file_data_of_challenge_phase = yaml_file_data['challenge_phases']
+            challenge_phase_ids = []
+            for data in yaml_file_data_of_challenge_phase:
+                serializer = ChallengePhaseSerializer(data=data, context={'challenge': challenge})
+                if serializer.is_valid():
+                    challenge_phase = serializer.save()
+                    challenge_phase_ids.append(challenge_phase.pk)
+
+                    test_annotation_file = data['test_annotation_file']
+                    if test_annotation_file:
+                        test_annotation_file_path = os.path.join("{}/{}/{}{}".format(BASE_LOCATION,
+                                                                                     unique_folder_name,
+                                                                                     extracted_folder_name,
+                                                                                     test_annotation_file))
+                    if os.path.isfile(test_annotation_file_path):
+                        challenge_test_annotation = open(test_annotation_file_path, 'rb')
+                        challenge_test_annotation_file = File(challenge_test_annotation)
+                        challenge_phase = ChallengePhase.objects.get(pk=challenge_phase.pk)
```

Comment:
Since this block of code is under transaction.atomic, it will automatically go into except block on line 571. So, should we still catch it here?